consensus/hotstuff/safetyrules/safety_rules_test.go

```diff
@@ -447,29 +483,75 @@ func (s *SafetyRulesTestSuite) TestProduceTimeout_ShouldTimeout() {
 // Hence, the cases tested here would be symptoms of an internal bugs, and therefore should not result in an NoVoteError.
 func (s *SafetyRulesTestSuite) TestProduceTimeout_NotSafeToTimeout() {
 
-	s.Run("highest-qc-below-locked-round", func() {
-		view := s.proposal.Block.View
-		newestQC := helper.MakeQC(helper.WithQCView(s.safetyData.LockedOneChainView - 1))
+	s.Run("newest-qc-nil", func() {
+		// newestQC cannot be nil
+		timeout, err := s.safety.ProduceTimeout(s.safetyData.LockedOneChainView, nil, nil)
+		require.Error(s.T(), err)
+		require.False(s.T(), model.IsNoVoteError(err))
```

Comment:
```suggestion
```
We never expect any sentinel from `ProduceTimeout`, per the [godoc](https://github.com/onflow/flow-go/blob/5f361374705d4565ce68b5a0ee2120ed83ceb047/consensus/hotstuff/safetyrules/safety_rules.go#L122). (We definitely wouldn't expect `NoVoteError`). So I don't think we need to check for absence of specific sentinels here.

Reply:
I think the interface specification is a bit outdated https://github.com/onflow/flow-go/blob/5f361374705d4565ce68b5a0ee2120ed83ceb047/consensus/hotstuff/safety_rules.go#L37-L43 
`NoTimeoutError` is no longer in the code base. Maybe we should just update the API and remove these checks?